python/cugraph/structure/graph_wrapper.pyx

```diff
@@ -193,20 +233,40 @@ def view_transposed_adj_list(graph_ptr):
 
     cdef uintptr_t offset_col_data = <uintptr_t> g.transposedAdjList.offsets.data
     cdef uintptr_t index_col_data = <uintptr_t> g.transposedAdjList.indices.data
-
-    offsets_data = rmm.device_array_from_ptr(offset_col_data,
-                                 nelem=offset_col_size,
-                                 dtype=np.int32)  # ,
-                                 # finalizer=rmm._make_finalizer(offset_col_data, 0))
-    indices_data = rmm.device_array_from_ptr(index_col_data,
-                                 nelem=index_col_size,
-                                 dtype=np.int32)  # ,
-                                 # finalizer=rmm._make_finalizer(index_col_data, 0))
-    # g.transposedAdjList.offsets.data and g.transposedAdjList.indices.data
-    # are not owned by this instance, so should not be freed here (this
-    # will lead to double free, and undefined behavior).
-
-    return cudf.Series(offsets_data), cudf.Series(indices_data)
+    cdef uintptr_t value_col_data = <uintptr_t> NULL
+    if g.transposedAdjList.edge_data is not NULL:
+        value_col_data = <uintptr_t> g.transposedAdjList.edge_data.data
+
+    # g.transposedAdjList.offsets.data, g.transposedAdjList.indices.data and
+    # g.transposedAdjList.edge_data.data are not owned by this instance, so
+    # should not be freed here (this will lead to double free, and undefined
+    # behavior).
+
+    offset_data = rmm.device_array_from_ptr(
+                       offset_col_data,
+                       nelem=offset_col_size,
+                       dtype=np.int32)  # ,
+                       # finalizer=rmm._make_finalizer(offset_col_data, 0))
+    offset_col = cudf.Series(offset_data)
+
+    index_data = rmm.device_array_from_ptr(
+                     index_col_data,
+                     nelem=index_col_size,
+                     dtype=np.int32)  # ,
+                     # finalizer=rmm._make_finalizer(index_col_data, 0))
```

Comment:
Is this dead code? Can this be removed?

Reply:
Yeah... I'd better move that to the comment saying this column data should not be freed by when python object becomes out-of-scope.

Reply:
Fixed.